lib/uuid_mixin.rb

```diff
@@ -2,6 +2,11 @@ module UuidMixin
   extend ActiveSupport::Concern
   included do
     before_validation :set_guid, :on => :create if respond_to?(:before_validation)
+    validates :guid,  :uniqueness => true, :presence => true
+  end
+
+  def dup
+    super.tap { |obj| obj.guid = nil }
```

Comment:
Curious how this will be affected by the default_value_for change in #18502 .  @bdunne perhaps you two should join forces and make a single PR?

Reply:
😭